govcd/vapp.go

```diff
@@ -786,6 +795,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
+		networkToUpdate.Configuration.ParentNetwork = &types.Reference{HREF: orgNetwork.HREF}
+	}
+	if networkToUpdate.Configuration.ParentNetwork != nil && orgNetwork == nil {
+		networkToUpdate.Configuration.ParentNetwork = nil
+	}
+	networkToUpdate.Description = networkSettingsToUpdate.Description
+	networkToUpdate.NetworkName = networkSettingsToUpdate.Name
+	networkToUpdate.Configuration.GuestVlanAllowed = networkSettingsToUpdate.GuestVLANAllowed
+	networkToUpdate.Configuration.IPScopes.IPScope[0].Gateway = networkSettingsToUpdate.Gateway
+	networkToUpdate.Configuration.IPScopes.IPScope[0].Netmask = networkSettingsToUpdate.NetMask
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNS1 = networkSettingsToUpdate.DNS1
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNS2 = networkSettingsToUpdate.DNS2
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNSSuffix = networkSettingsToUpdate.DNSSuffix
+	networkToUpdate.Configuration.IPScopes.IPScope[0].IPRanges = &types.IPRanges{IPRange: networkSettingsToUpdate.StaticIPRanges}
+
+	// for case when range is one ip address
+	if networkSettingsToUpdate.DhcpSettings != nil && networkSettingsToUpdate.DhcpSettings.IPRange != nil && networkSettingsToUpdate.DhcpSettings.IPRange.EndAddress == "" {
+		networkSettingsToUpdate.DhcpSettings.IPRange.EndAddress = networkSettingsToUpdate.DhcpSettings.IPRange.StartAddress
+	}
+
+	// remove DHCP config
+	if networkSettingsToUpdate.DhcpSettings == nil {
+		networkToUpdate.Configuration.Features.DhcpService = nil
+	}
+
+	// create DHCP config
+	if networkSettingsToUpdate.DhcpSettings != nil && networkToUpdate.Configuration.Features.DhcpService == nil {
+		networkToUpdate.Configuration.Features.DhcpService = &types.DhcpService{
+			IsEnabled:        networkSettingsToUpdate.DhcpSettings.IsEnabled,
+			DefaultLeaseTime: networkSettingsToUpdate.DhcpSettings.DefaultLeaseTime,
+			MaxLeaseTime:     networkSettingsToUpdate.DhcpSettings.MaxLeaseTime,
+			IPRange:          networkSettingsToUpdate.DhcpSettings.IPRange}
+	}
+
+	// update DHCP config
+	if networkSettingsToUpdate.DhcpSettings != nil && networkToUpdate.Configuration.Features.DhcpService != nil {
+		networkToUpdate.Configuration.Features.DhcpService.IsEnabled = networkSettingsToUpdate.DhcpSettings.IsEnabled
+		networkToUpdate.Configuration.Features.DhcpService.DefaultLeaseTime = networkSettingsToUpdate.DhcpSettings.DefaultLeaseTime
+		networkToUpdate.Configuration.Features.DhcpService.MaxLeaseTime = networkSettingsToUpdate.DhcpSettings.MaxLeaseTime
+		networkToUpdate.Configuration.Features.DhcpService.IPRange = networkSettingsToUpdate.DhcpSettings.IPRange
+	}
+
+	currentNetworkConfiguration.NetworkConfig[networkToUpdateIndex] = networkToUpdate
+
+	return updateNetworkConfigurations(vapp, currentNetworkConfiguration.NetworkConfig)
+}
+
+// UpdateOrgNetwork updates to vApp added org VDC network
```

Comment:
Maybe this would be better:
```suggestion
// UpdateOrgNetwork updates Org VDC network which is part of a vApp
```

Reply:
updated